iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -249,6 +228,34 @@ private async void ConnectionStatusChangeHandler(ConnectionStatus status, Connec
             }
         }
 
+        private async Task GetTwinOnConnectAsync(CancellationToken cancellationToken)

```

Comment:
I think the method name is misleading because it could be called any time. I think where you call it and a comment there is all that you need.

So, FWIW, I'd just name it `GetTwinAndDetectChanges()` or something like that, but just my opinion.